libs/blocks/global-navigation/utilities/utilities.js

```diff
@@ -122,50 +122,68 @@ export function decorateCta({ elem, type = 'primaryCta', index } = {}) {
     </div>`;
 }
 
-export function closeAllDropdowns({ e } = {}) {
-  const openElements = document.querySelectorAll(`${selectors.globalNav} [aria-expanded='true']`);
+let curtainElem;
+export function setCurtainState(state) {
+  if (typeof state !== 'boolean') return;
+
+  curtainElem = curtainElem || document.querySelector(selectors.curtain);
+  if (curtainElem) curtainElem.classList.toggle('feds-curtain--open', state);
+}
+
+export const isDesktop = window.matchMedia('(min-width: 900px)');
+
+export function setActiveDropdown(elem) {
+  const activeClass = selectors.activeDropdown.replace('.', '');
+
+  // We always need to reset all active dropdowns at first
+  const resetActiveDropdown = () => {
+    [...document.querySelectorAll(selectors.activeDropdown)]
+      .forEach((activeDropdown) => activeDropdown.classList.remove(activeClass));
+  };
+  resetActiveDropdown();
+
+  // If no elem is provided, de-activating all dropdowns is enough
+  if (!(elem instanceof HTMLElement)) return;
+
+  // Compose an array of parents that could be active dropdowns
+  const selectorArr = [selectors.menuSection, selectors.menuColumn, selectors.navItem];
+
+  // Look for the first parent that fits the active dropdown criteria
+  while (selectorArr.length) {
+    const closestSection = elem.closest(selectorArr.shift());
+
+    if (closestSection && closestSection.querySelector('[aria-expanded = "true"]')) {
+      closestSection.classList.add(activeClass);
+      break;
+    }
+  }
```

Comment:
nit - use `some`:
```suggestion
selectorArr.some((selector) => {
  const closestSection = elem.closest(selector);

  if (closestSection && closestSection.querySelector('[aria-expanded="true"]')) {
    closestSection.classList.add(activeClass);
    return true;
  }

  return false;
});
```

Reply:
Great suggestion! I wanted a better way than using `while`, but `some` just slipped my mind. I've implemented it in the new commit.